src/main/groovy/lang/ObjectRange.java

```diff
@@ -488,37 +500,61 @@ public void remove() {
 
     @Override
     public List step(int step) {
+        return step((Number) step);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List step(Number step) {
         final IteratorClosureAdapter adapter = new IteratorClosureAdapter(this);
         step(step, adapter);
         return adapter.asList();
     }
 
     /**
-     * Increments by one
+     * Increments by step size
      *
      * @param value the value to increment
-     * @return the incremented value
+     * @param step  the value to increment by
+     * @return the incremented value or null, if there isn't any
      */
-    private static Object increment(Object value) {
+    protected static Comparable increment(Comparable value, Number step) {
```

Comment:
because it got messed up with rebases (changed to private and when I changed the order of commits, it got changed back):
https://github.com/apache/groovy/pull/360/files#diff-edcd680998449caffc50809917864ceeR523

Thanks, fixing it now.